task-processor/task-manager/task_manager_wrapper.go

```diff
@@ -0,0 +1,146 @@
+package taskmanager
+
+import (
+	"context"
+
+	"github.com/Layr-Labs/eigensdk-go/chainio/txmgr"
+	internalutils "github.com/Layr-Labs/eigensdk-go/internal/utils"
+	sdktypes "github.com/Layr-Labs/eigensdk-go/types"
+	"github.com/Layr-Labs/eigensdk-go/utils"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+var _ TaskManagerContract[any, any] = (*TaskManagerContractWrapper[any, any])(nil)
+
+type TaskManagerContractWrapper[Input any, Output any] struct {
```

Comment:
Let's make this private:

```suggestion
var _ TaskManagerContract[any, any] = (*TaskManagerContractWrapper[any, any])(nil)

type taskManagerContractWrapper[Input any, Output any] struct {
```

Reply:
Done in [b26edbb](https://github.com/Layr-Labs/eigensdk-go/pull/679/commits/b26edbb61490c81c9a5ef62ebf6f6e7677c07a6d)